components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/util/Utils.java

```diff
@@ -1128,6 +1128,38 @@ private static String generateSMSOTP(String charSet, int otpLength) {
         return sb.toString();
     }
 
+    /**
+     * Return user account state.
+     *
+     * @param user  User.
+     * @return account state.
+     */
+    public static String getAccountState(User user) {
+
+        String accountState = StringUtils.EMPTY;
+        try {
+            org.wso2.carbon.user.core.UserStoreManager userStoreManager = IdentityRecoveryServiceDataHolder.
+                    getInstance().getRealmService().getBootstrapRealm().getUserStoreManager();
+            while (!userStoreManager.isExistingUser(user.getUserName())) {
+                userStoreManager = userStoreManager.getSecondaryUserStoreManager();
+                if (userStoreManager == null) {
+                    return accountState;
+                }
+            }
+            Map<String, String> claimMap =
+                    ((AbstractUserStoreManager) userStoreManager).getUserClaimValues(user.getUserName(),
+                            new String[]{IdentityRecoveryConstants.ACCOUNT_STATE_CLAIM_URI}, "default");
+            if (!claimMap.isEmpty()) {
+                if (claimMap.containsKey(IdentityRecoveryConstants.ACCOUNT_STATE_CLAIM_URI)) {
+                    accountState = claimMap.get(IdentityRecoveryConstants.ACCOUNT_STATE_CLAIM_URI);
+                }
+            }
+        } catch (org.wso2.carbon.user.core.UserStoreException e) {
+            log.error("Error occurred while retrieving UserStoreManager");
```

Comment:
Since we are logging error we have to print the stack trace as well. 
```suggestion
            log.error("Error occurred while retrieving UserStoreManager", e);
```